src/Repl/Decode.test.tsx

```diff
@@ -0,0 +1,271 @@
+/*
+ * BoGLServerRequest.test.js
+ * Testing for BoGL server requests
+ */
+
+import * as d from './Decode'
+
+it('Test decode Ints', () => {
+  expect(d.decodeValue({
+    type: "Int",
+    value: 123
+  })).toEqual(123)
+
+  expect(d.decodeValue({
+    type: "Int",
+    value: -23534543
+  })).toEqual(-23534543)
+
+  expect(d.decodeValue({
+    type: "Int",
+    value: 0
+  })).toEqual(0)
+
+  expect(d.decodeValue({
+    type: "Int",
+    value: 987653423
+  })).toEqual(987653423)
+});
+
+
+it('Test decode Bools', () => {
+  expect(d.decodeValue({
+    type: "Bool",
+    value: true
+  })).toEqual("True")
+
+  expect(d.decodeValue({
+    type: "Bool",
+    value: false
+  })).toEqual("False")
+
+  // any other val will be coerced to a bool...
```

Comment:
why?

Reply:
Ah so this relates back to the response from the server. All bool values are returned in the form:
```json
{
    "type": "Bool",
    "value": true/false
}
```
The way the response is structured from the backend, the bools end up becoming literals, so decoding them into strings is necessary for that reason. However if there's anything else but a Bool in the value, it's because the server sent something unexpected. Usually if this happens it's not a parseable response in the first place. But I don't think a non-bool coming back as a Bool is possible due to the nature of the API being in haskell. I think if you tried to return something that was of type Bool but not a bool, it would crash haskell instead before it ever got here.

So far I've never observed anything else besides `true/false` coming back, but it was something that I realized should be noted. The alternative is to play it strict and fail on any non-bool value, which is a possibility too. If we do that, we should be strict with all other categories of values as well.

Reply:
I went ahead and added an exception thrown for this. It should never bit hit, but in the off chance it does, it will probably be because something is messed up in the response.